cpp/dolfinx/geometry/GJK.cpp

```diff
@@ -0,0 +1,215 @@
+// Copyright (C) 2020 Chris Richardson
+//
+// This file is part of DOLFINX (https://www.fenicsproject.org)
+//
+// SPDX-License-Identifier:    LGPL-3.0-or-later
+
+#include "GJK.h"
+#include <Eigen/Geometry>
+#include <array>
+#include <iomanip>
+#include <iostream>
+#include <stdexcept>
+
+using namespace dolfinx;
+
+namespace
+{
+std::pair<Eigen::Matrix<double, Eigen::Dynamic, 3, Eigen::RowMajor>,
```

Comment:
I know. It could alternatively return a fixed size (we know the max number of rows), and an int, indicating the number of rows, but that seemed more messy.

Reply:
Eigen has a template argument for the 'max' rows/cols to handle cases like this without dynamic memory allocation.